homeassistant/components/media_player/kodi.py

```diff
@@ -157,26 +158,46 @@ def _check_deprecated_turn_off(hass, turn_off_action):
 def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
     """Set up the Kodi platform."""
     if DATA_KODI not in hass.data:
-        hass.data[DATA_KODI] = []
-    name = config.get(CONF_NAME)
-    host = config.get(CONF_HOST)
-    port = config.get(CONF_PORT)
-    tcp_port = config.get(CONF_TCP_PORT)
-    encryption = config.get(CONF_PROXY_SSL)
-    websocket = config.get(CONF_ENABLE_WEBSOCKET)
-
-    entity = KodiDevice(
-        hass,
-        name=name,
-        host=host, port=port, tcp_port=tcp_port, encryption=encryption,
-        username=config.get(CONF_USERNAME),
-        password=config.get(CONF_PASSWORD),
-        turn_on_action=config.get(CONF_TURN_ON_ACTION),
-        turn_off_action=config.get(CONF_TURN_OFF_ACTION),
-        timeout=config.get(CONF_TIMEOUT), websocket=websocket)
-
-    hass.data[DATA_KODI].append(entity)
-    async_add_devices([entity], update_before_add=True)
+        hass.data[DATA_KODI] = dict()
+
+    # Is this a manual configuration?
+    if config.get(CONF_HOST) is not None:
+        name = config.get(CONF_NAME)
+        host = config.get(CONF_HOST)
+        port = config.get(CONF_PORT)
+        tcp_port = config.get(CONF_TCP_PORT)
+        encryption = config.get(CONF_PROXY_SSL)
+        websocket = config.get(CONF_ENABLE_WEBSOCKET)
+    elif discovery_info is not None:
+        name = "{} ({})".format(DEFAULT_NAME, discovery_info.get('hostname'))
+        host = discovery_info.get('host')
+        port = discovery_info.get('port')
+        tcp_port = DEFAULT_TCP_PORT
+        encryption = DEFAULT_PROXY_SSL
+        websocket = DEFAULT_ENABLE_WEBSOCKET
+    else:
+        _LOGGER.warning("Cannot determine device")
+        return
+
+    # Only add a device once, so discovered devices do not override manual
+    # config.
+    ip_addr = socket.gethostbyname(host)
+    if ip_addr not in hass.data[DATA_KODI]:
```

Comment:
Just return. 

```python
if ip_addr in hass.data[DATA_KODI]:
    return
```

Reply:
Same as above.

Reply:
Ahh OK, now I can see your point. The return is missing for else. But why shouldn't there be a log message?

Reply:
The log message should be removed or at least be downgraded to `debug`. It is already logging that the entity is added.